### PR TITLE
[chore] Make Dockerignore more aggresive.

### PR DESCRIPTION
**Pull request by EpicPi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #84
* #76
* #75
* #74
* #73
* #72
* __->__ #71
* #85

## Changes made
Add more files to dockerignore.

## Testing
- [x] End-to-End

## Work left to be done

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,5 +1,9 @@
-client/node_modules
-client/build
+# Used to trim context during during docker builds.
```
Comment (SammyBaek) [inline .dockerignore:1]:
```suggestion
# Used to trim context during docker builds.
```

**Review by SammyBaek — APPROVED** [review]
ELI5 plz

File: .dockerignore (hunk 1 above)
Comment (EpicPi) [reply .dockerignore]: So, when you do `docker build` the only positional argument it takes is the context. The context is the working directory on the host machine. `.dockerignore` is only used during this `docker build` process.

Docker copies all the files in the context into a temporary file location and then starts executing the Dockerfile. This is useful in cases when you want to copy files from the host machine into the image during build. 

We have no reason to copy anything other than what's in server/ since all we use containers for right now is to run flask.
